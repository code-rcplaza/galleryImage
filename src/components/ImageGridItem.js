import React from 'react'
import PropTypes from 'prop-types'
import { capitalizeFirstLetter } from '../helpers/capitalizeFirstLetter'

const ImageGridItem = ({ title, url }) => {
    
    return (
        <a href={url} rel="noopener noreferrer" target="_blank">
            <div className="img-container animate__animated animate__fadeIn animate__delay-0.5s">
                <img className="img-item" src={url} alt="img" />
                <div className="line-container">
                    <span className="single-line">{capitalizeFirstLetter(title)}</span>
                </div>
            </div>
        </a>
    )
}


ImageGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}

export default ImageGridItem

