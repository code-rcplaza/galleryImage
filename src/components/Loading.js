import React from 'react'
import { capitalizeFirstLetter } from '../helpers/capitalizeFirstLetter'

export const Loading = () => {

    return (
        <div className="loading">
            <div className="loader-container">
                <div className="single6"></div>
            </div>
            
            <div className="line-container">
                <span className="single-line">{capitalizeFirstLetter('Loading')}</span>
            </div>
        </div>
    )
}
