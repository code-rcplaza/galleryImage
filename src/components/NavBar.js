import React from 'react'

export const NavBar = () => {
    return (
       <header>
           <h2>Sky Test</h2>
       </header>
    )
}
