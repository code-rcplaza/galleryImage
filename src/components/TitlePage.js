import React from 'react'

export const TitlePage = () => {
    return (
        <div className="container-sub-header">
            <h1 className="title-page">This a image gallery</h1>
            <p className="sub-title">By Ricardo Plaza</p>
        </div>
    )
}
