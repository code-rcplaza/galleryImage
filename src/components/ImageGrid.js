// import React, { useEffect, useState } from 'react'
import React from 'react';
import PropTypes from 'prop-types'
import { useFetchImgs } from '../hooks/useFetchImgs';
import ImageGridItem from './ImageGridItem'
import { Loading } from './Loading';

const ImageGrid = ({ category }) => {

    const { data: imgs, loading } = useFetchImgs(category);

    return (
        <>
            {loading ? <Loading /> :

                <div className="container">
                    {
                        imgs.map(({ id, title, url }) => (
                            <ImageGridItem key={id} url={url} title={title} />)
                        )
                    }
                </div>
            }
        </>
    )
}

ImageGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default ImageGrid


