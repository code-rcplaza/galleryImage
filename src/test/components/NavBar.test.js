import React from 'react';
import { shallow } from "enzyme";
import { NavBar } from "../../components/NavBar";

describe('Test on <NavBar />', () => {

    test('Should be correctly render', () => {
        const wrapper = shallow(<NavBar /> )

        expect( wrapper ).toMatchSnapshot()
    });
});