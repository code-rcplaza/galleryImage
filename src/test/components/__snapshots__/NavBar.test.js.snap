// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <NavBar /> Should be correctly render 1`] = `
<header>
  <h2>
    Sky Test
  </h2>
</header>
`;
